src/lib/y2network/clients/dns.rb

```diff
@@ -0,0 +1,284 @@
+# Copyright (c) [2020] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2network/config"
+
+Yast.import "DNS"
+Yast.import "Label"
+Yast.import "Lan"
+Yast.import "NetworkService"
+Yast.import "Wizard"
+
+Yast.import "CommandLine"
+Yast.import "RichText"
+
+module Y2Network
+  module Clients
+    # DNS client for configuring DNS and hostname settings
+    class DNS < Yast::Client
+      include Yast::Logger
+
+      attr_reader :hostname, :nameserver1, :nameserver2, :nameserver3
+
+      # Constructor
+      def initialize
+        textdomain "network"
+        Yast.include self, "network/services/dns.rb"
+
+        @hostname = "hostname"
+        @nameserver1 = "nameserver1"
+        @nameserver2 = "nameserver2"
+        @nameserver3 = "nameserver3"
+      end
+
+      def main
+        log_and_return { CommandLine.Run(cmdline_definition) }
+      end
+
+    private
+
+      # Main DNS GUI
+      def DNSGUI
+        Wizard.CreateDialog
+        Wizard.SetDesktopTitleAndIcon("dns")
+        read_config
+
+        Wizard.SetNextButton(:next, Label.FinishButton)
+
+        # main ui function
+        ret = DNSMainDialog(true)
+        log.debug("ret == #{ret}")
+
+        if (ret == :next) && modified?
+          write_config
+
+          # no more workarounds with dhcp-clients
+          # do a full network restart (bnc#528937)
+          Yast::NetworkService.StartStop
+        end
+
+        UI.CloseDialog
+        ret
+      end
+
+      def modified?
+        Yast::Lan.system_config != Yast::Lan.yast_config
+      end
+
+      def valid_hostname?(value)
+        value.empty? || Yast::Hostname.Check(value.tr(".", ""))
+      end
+
+      def config
+        Yast::Lan.yast_config
+      end
+
+      def log_and_return(&block)
+        # The main ()
+        log.info("----------------------------------------")
+        log.info("Dns module started")
+        ret = block.call
+        log.debug("ret=#{ret}")
+        # Finish
+        log.info("Dns module finished")
```

Comment:
well ret is usually next or abort and I find quite annoying to not see it in log. So I would remove that log.debug and instead use `        log.info("Dns module finished(#{ret.inspect})")`